pkg/forwarder/backoff_test.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package forwarder
+
+import (
+	"math"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
You should actually seed with a known value so your tests are always deterministic and do not depend on when you run them

Reply:
You can actually do those in tests (mark them as non-parallel if you want) so you really control your seed

Reply:
We want our results to be random to truly test our algorithm I believe, right?

Reply:
Here you introduce potential flakiness and not random test.
The problem that will happen here if if this test fails, people will retry once and it will works so they will think there is just flakiness.
If you want to "truly" test the algorithm, you can just have a lot of numbers of tests, i.e: in your test add a `for i := 0; i < 1000; i++ {}` to run your test on 1000 different seeds.

If you even want more than that, you can put it in a benchmark with `for i := 0, i < b.N; i++ {}`
and then run something like `-bench . -benchtime=5m` to thoroughly test.